src/main/java/org/junitpioneer/jupiter/PioneerAnnotationUtils.java

```diff
@@ -117,6 +121,39 @@ public static boolean isAnyRepeatableAnnotationPresent(ExtensionContext context,
 		return findAnnotations(context, annotationType, true, true);
 	}
 
+	/**
+	 * Returns the annotations <em>present</em> on the {@code AnnotatedElement}
+	 * that are annotated with the specified annotation. The meta-annotation can be <em>present</em>,
+	 * <em>indirectly present</em>, <em>meta-present</em>, or <em>enclosing present</em>.
+	 */
+	public static <A extends Annotation> List<Annotation> findAnnotatedAnnotations(AnnotatedElement element,
+			Class<A> annotation) {
+		return Arrays
+				.stream(element.getDeclaredAnnotations())
+				// flatten @Repeatable aggregator annotations
+				.flatMap(PioneerAnnotationUtils::flatten)
+				.filter(a -> !(findOnElement(a.annotationType(), annotation, true).isEmpty()))
+				.collect(Collectors.toList());
+	}
```

Comment:
Without looking into this, I'm a but surprised that there's a combination of annotation search parameters that we don't already cover. Can you help me understand how this is different from the other methods?

Reply:
No. 👿

Reply:
JUnits `AnnotationSupport` and our own `PioneerAnnotationUtils` has a lot of methods for getting specific annotations _directly_. We can not get annotations based on a criteria. I wanted to get all annotations on a test method that are themselves annotated with `@ArgumentsSource`. So, that's what this does - returns all annotations on an `AnnotatedElement` that are *annotated with a given annotation*.